ddtrace/internal/ci_visibility/api/_session.py

```diff
@@ -78,10 +92,45 @@ def _telemetry_record_event_finished(self):
             test_framework=self._session_settings.test_framework_metric_name,
             has_codeowners=self._codeowners is not None,
             is_unsupported_ci=self._session_settings.is_unsupported_ci,
+            early_flake_detection_abort_reason="faulty" if self.efd_is_faulty_session() else None,
```

Comment:
We have `self._efd_is_faulty_session` and `self.efd_is_faulty_session()`. Above the first one was used, here the second one was. Should they be the same?

The method computes the property, right? Maybe we should make the method return the property value immediately if it's already set, and use the method everywhere.

Reply:
Yep. There's only one other place where we were looking at the attribute directly, so I updated that in e796da1000be1acf2980de8a3000061553bb1092 .